app/src/main/java/com/rentminder/service/MembersService.kt

```diff
@@ -9,11 +9,18 @@ import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
 class MembersService {
-    suspend fun fetchMembersList() : List<Members> {
-        return withContext(Dispatchers.IO){
-            val service = RetrofitClientInstance.retrofitInstance?.create(MembersDAO::class.java)
-            val members = async {service?.getAllMembers()?.awaitResponse()?.body() }
-            return@withContext members.await()?: emptyList()
+    private val service: MembersDAO by lazy {
+        RetrofitClientInstance.retrofitInstance?.create(MembersDAO::class.java)
+            ?: throw IllegalStateException("RetrofitClientInstance is null")
+    }
+
+    suspend fun fetchMembersList(): List<Members> {
+        return withContext(Dispatchers.IO) {
+            try {
+                service.getAllMembers().awaitResponse().body() ?: emptyList()
+            } catch (e: Exception) {
+                emptyList()
+            }
```

Comment:
Added try catch block here to ensure that no exceptions are going to crash the app. If there is an exception, instead of crashing it will return an 'emptyList()'

Reply:
Ditto comment above.